packages/lisk-transactions/src/0_transfer_transaction.ts

```diff
@@ -179,9 +179,9 @@ export class TransferTransaction extends BaseTransaction {
 	protected applyAsset(store: StateStore): ReadonlyArray<TransactionError> {
 		const errors: TransactionError[] = [];
 		const sender = store.account.get(this.senderId);
-
 		const balanceError = verifyBalance(this.id, sender, this.amount);
-		if (balanceError) {
+		// Only return error if account has enough funds for fee to avoid duplicate errors
+		if (balanceError && new BigNum(sender.balance).gte(0)) {
```

Comment:
maybe check if the sender balance first before executing `verifyBalance`, and we could add simple function in utils/verify.ts to capture repeating `new BigNum(sender.balance).gte(0)`?
Also, this wouldn't solve the problem with the error message when sender has fee, but not amount case ='(.